libraries/chain/include/eosio/chain/hotstuff.hpp

```diff
@@ -41,8 +77,13 @@ namespace eosio::chain {
       quorum_certificate_message          justify; //justification
       uint8_t                             phase_counter = 0;
 
+      digest_type get_proposal_id() const { return get_digest_to_sign(block_id, phase_counter, final_on_qc); };
+
       uint32_t block_num() const { return block_header::num_from_id(block_id); }
-      uint64_t get_height() const { return compute_height(block_header::num_from_id(block_id), phase_counter); };
+      uint64_t get_key() const { return compute_height(block_header::num_from_id(block_id), phase_counter); };
+
+      view_number get_view_number() const { return view_number(std::make_pair(block_header::num_from_id(block_id), phase_counter)); };
```

Comment:
As Greg already pointed out, you don't need the `pair` constructor, just use the constructor that takes the two values.